lightning/src/routing/network_graph.rs

```diff
@@ -562,9 +562,14 @@ impl NetworkGraph {
 		}
 	}
 
-	/// For an already known node (from channel announcements), update its stored properties from a given node announcement
+	/// For an already known node (from channel announcements), update its stored properties from a given node announcement.
+	///
+	/// You probably don't want to call this directly, instead relying on a NetGraphMsgHandler's
+	/// RoutingMessageHandler implementation to call it indirectly. This may be useful to accept
+	/// routing messages without checking their signatures.
+	///
 	/// Announcement signatures are checked here only if Secp256k1 object is provided.
-	fn update_node_from_announcement(&mut self, msg: &msgs::NodeAnnouncement, secp_ctx: Option<&Secp256k1<secp256k1::VerifyOnly>>) -> Result<bool, LightningError> {
+	pub fn update_node_from_announcement<T: secp256k1::Verification>(&mut self, msg: &msgs::NodeAnnouncement, secp_ctx: Option<&Secp256k1<T>>) -> Result<bool, LightningError> {
```

Comment:
Hm, seems more cautious to make this method to the effect of `pub(feature = fuzztesting)` for now, same below. Because it seems like we'll generally want users to use `handle_node_announcement` and similar, so seems more likely to be accidentally used until we get a user request for it.

Reply:
Right. I mean its not an ideal API, but IIRC it was intended to be pub initially - there's a strong argument for wanting to process messages without verifying signatures for users who may be fetching verified/trusted routing graph updates.

One thing we could do, alternatively, is swap the `msg` here for the unsigned `contents` version, but I do kinda like having the option to verify a signature, even if its a somewhat dubious API for that.

Reply:
> One thing we could do, alternatively, is swap the msg here for the unsigned contents version,

Hmm, I somewhat prefer that, just to have a clearer public API/there's only one "right way" for each path. I guess not a huge deal though.

Reply:
Right, I guess I see it more as "`NetGraphMsgHandler` just happens to implement the message handling trait, but you can happily pass it in messages you got from elsewhere for various reasons". I admit that API isn't perfect, though if anything the change that we'd want is just to move the UTXO-checking logic from `handle_channel_announcement` here.

Reply:
I'll follow up on this in a separate PR.